### PR TITLE
Add `history` builtin command

### PR DESCRIPTION
**Pull request by x4lldux**
Dumps the history file with numbered lines.

### CODE REVIEW

File: src/sh.janet [hunk 1]
```diff
@@ -1102,6 +1102,29 @@
...
 
+(defn- make-history-builtin
+  []
+  @{
+    :pre-fork
+      (fn builtin-history [self args] nil)
+    :post-fork
+      (fn builtin-history
+        [self args]
+        (let [hist-file (-> (fiber/current)
```
Comment (x4lldux) [inline src/sh.janet:1113]: @andrewchambers please check this. feels hacky, but couldn't find a different way.

Comment (andrewchambers) [reply src/sh.janet:1113]: I think the correct way is to use the history api from readline - https://tiswww.case.edu/php/chet/readline/history.html#SEC6

So it requires some C programming, or we rewrite our own version of readline in janet.

Comment (x4lldux) [reply src/sh.janet:1113]: Will it work with libnoise or libedit?

Comment (andrewchambers) [reply src/sh.janet:1113]: libedit follows the same api yes. libnoise it needs to be written, or just do nothing, (janet error may be ok) but still compile.

**Comment by andrewchambers** [comment]
Closing for now, I think it really needs to use the history api, which will be completely different.

**Pull request closed without merging**